R/bundleFiles.R

```diff
@@ -0,0 +1,246 @@
+    } else {
+      # must be an ordinary file
+      exploded <- c(exploded, f)
+    }
+  }
+
+  size <- sum(file.info(file.path(dir, exploded))$size, na.rm = TRUE)
+  enforceBundleLimits(dir, size, length(exploded))
+
+  exploded
+}
+
+#' List Files to be Bundled
+#'
+#' Given a directory containing an application, returns the names of the files
+#' to be bundled in the application.
+#'
+#' @param appDir Directory containing the application.
+#'
+#' @details This function computes results similar to a recursive directory
+#' listing from [list.files()], with the following constraints:
+#'
+#' \enumerate{
+#' \item{If the total size of the files exceeds the maximum bundle size, no
+#'    more files are listed. The maximum bundle size is controlled by the
+#'    `rsconnect.max.bundle.size` option.}
+#' \item{If the total size number of files exceeds the maximum number to be
+#'    bundled, no more files are listed. The maximum number of files in the
+#'    bundle is controlled by the `rsconnect.max.bundle.files` option.}
+#' \item{Certain files and folders that don't need to be bundled, such as
+#'    those containing internal version control and RStudio state, are
+#'    excluded.}
+#' \item{In order to stop specific files in the working directory from being
+#'    listed in the bundle, the files must be listed in the .rscignore file.
+#'    This file must have one file or directory per line with no support for
+#'    wildcards.}
+#' }
+#'
+#' @return Returns a list containing the following elements:
+#'
+#' \tabular{ll}{
+#' `contents` \tab A list of the files to be bundled \cr
+#' `totalSize` \tab The total size of the files \cr
+#' }
+#'
+#' @export
+listBundleFiles <- function(appDir) {
+  paths <- recursiveBundleFiles(appDir)
+
+  files <- length(paths)
+  size <- sum(file.info(file.path(appDir, paths))$size)
+
+  enforceBundleLimits(appDir, size, files)
+  list(
+    totalSize = size,
+    totalFiles = files,
+    contents = paths
+  )
+}
+
+bundleFiles <- function(appDir) {
+  listBundleFiles(appDir)$contents
+}
+
+recursiveBundleFiles <- function(dir, depth = 0) {
+  # generate a list of files at this level
+  contents <- list.files(dir, all.files = TRUE, no.. = TRUE, include.dirs = TRUE)
+
+  # exclude some well-known files/directories at root level
+  if (depth == 0) {
+    contents <- contents[!grepl(glob2rx("*.Rproj"), contents)]
+    contents <- setdiff(
+      contents,
+      c("manifest.json", "rsconnect", "packrat", "app_cache", ".Rproj.user")
+    )
+  }
+
+  # exclude renv files, knitr cache dirs, and another well-known files
+  contents <- setdiff(contents, c("renv", "renv.lock"))
+  contents <- contents[!isKnitrCacheDir(contents)]
+  contents <- setdiff(
+    contents,
+    c(".DS_Store", ".git", ".gitignore", ".quarto", ".Rhistory", ".svn")
+  )
+
+  # remove any files lines listed .rscignore
+  if (".rscignore" %in% contents) {
+    ignoreContents <- readLines(file.path(dir, ".rscignore"))
```

Comment:
Two changes here:

* `readLines(file.path(dir, ".rscignore"))` instead of `readLines(".rscignore")`; if that ever worked  before it was by coincidence
* Also remove `.rscignore` itself from the bundle

Reply:
Likely worked because:
* ignore files usually at top level
* working directory almost always the content directory